### ISSUE TITLE
Return item quest crash

### ISSUE THREAD

**Issue by litvyakov1488** [Decane/SRP#13]
Hello Decane. I hope you still support your Stalker CS patch. I try to enjoy the game from a fresh start since its almost bug free now but I stumble upon an really annoying bug. I want so desperately to fix that so I even signed up on this GitHub thing to report this bug. Anyway, I get to the point.
At the Cordon location there are several unique optional “return item” quests. One of them in particular requires to return the magic vodka to Loners. Whenever I try to get the quest via dialog my game crashes to desktop without any errors or “submit a bug window”. Only “xr_engine.exe” stopped working pop-up. That’s it.
There is something I’ve done before trying to get the quest - I’ve looted the body near train car which is supposed to have the quest related vodka on it. Hope this helps.

I use SRP 1.1.3, without any optional mods. Just clean steam Rus version of stalker cs and srp 1.1.3 with widescreen fix and new shaders (to fix the FPS drops on modern hardware). Hope you’ll help me out. Thanks.

**Comment by Decane** [comment]
Hi, can you please:

1. Upload a save file from which I can reproduce the crash.
2. Tell me which version 1.1.3 you are using - is it [one of the release candidates](https://github.com/Decane/SRP/releases), or are you using a version of the master branch from a time that postdates beta 2?
3. Post the last couple of lines from your crashlog (see 'Troubleshooting' section of the SRP readme).

**Comment by litvyakov1488** [comment]
1.
[uriel_quicksave.zip](https://github.com/Decane/SRP/files/1878122/uriel_quicksave.zip)
done.

**Comment by litvyakov1488** [comment]
2. I use ver. from the master brunch section (https://github.com/Decane/SRP)

**Comment by litvyakov1488** [comment]
3. FATAL ERROR
 
[error]Expression    : fatal error
[error]Function      : xray::core::detail::strconcat_error::process
[error]File          : E:\priquel\sources\engine\xrCore\string_concatenations.cpp
[error]Line          : 34
[error]Description   : <no expression>
[error]Arguments     : buffer overflow: cannot concatenate strings(1):

[d:\steam\steamapps\common\stalker clear sky\gamedata\sounds\][

characters_voice\dialogs\Жил у нас тут в Зоне химик один. Не то, чтобы в Зоне химией занимался или алхимией какой, но «волшебную водку» готовил на ура. Словно целебный напиток... Выпьешь и сразу забываешь про всё: про Зону, про мутантов, про аномалии. Тебе уже ничего не страшно, и все равно кто ты и что с тобой, вставляет как никогда. За эту водку такие цены набивали, мама не горюй. До того дошли, что химика решили отравить толи из-за зависти, толи из-за жадности. Не себе, не людям, как говорится. Отравили, не.ogg]

**Comment by Decane** [comment]
@litvyakov1488 Thanks, I've reproduced the crash and isolated the culprit. It's one of my recent code changes. I'll close this issue once I've pushed a fix to master.

**Comment by litvyakov1488** [comment]
Thanks a lot. That was really fast. Godspeed, Decane!

**Comment by litvyakov1488** [comment]
So in order to fix my issue I need to download recently updated “gamedata” folder from the master brunch section and copy it to my CS game location, right?

**Comment by Decane** [comment]
Yes, that would be easiest. You could also manually make the changes I made in commit 6829955 to the version of task_objects.script in your gamedata folder, but I'd recommend the former option. Note: If you use `git pull` to update your gamedata, there shouldn't be a need to re-do the widescreen patch and shader mod you are using. Otherwise, you'll probably want to re-apply those after updating. (If you have no idea what `git pull`, just ignore the previous sentence.)

What you should **not** do is just copy the latest task_objects.script from the master branch over the one you have without also updating the other recently changed scripts, as they have cross-dependencies.

**Comment by litvyakov1488** [comment]
Again a trillion thanks to you for so quick respond. I’ll try the fix first thing in the morning.
